### PR TITLE
Feat: nomination commission

### PR DESCRIPTION
**Pull request by sander2**
Implement commission: every time we withdraw from the global pool, we directly transfer the commission to the operator. The remaining goes into the vault's staking pool. That also means that there's no technical reason to place restrictions on when the commission can be modified, although there could be other reasons to add such restrictions in the future. 

One drawback of this approach, though, is that it is less clear for the operator to see how much they have earned.

I also added a randomized test to test the reward distribution, which compares the actual rewards with one calculated by a very simple (and therefore less error-prone) brute-force implementation. I am aware that fee distribution is going to change in the future but the test should be easy enough to update when that happens.

Also note the fix in the vault-registry that prevents incorrect reward distribution after vault collateral changes.

A final thing to note is that the commission is stored and set in the fee pallet rather than the nomination pallet. There is an argument to be made to place it in the nomination pallet, but that'd slightly complicate the implementation (we'd require a config item trait that allows the fee pallet to make calls to the nomination pallet. But in the end the commission _is_ directly related to the fees so it's not a terrible spot for it either.

### CODE REVIEW

File: standalone/runtime/tests/test_redeem.rs [hunk 1]
```diff
@@ -35,11 +35,11 @@ fn test_with<R>(execute: impl Fn(VaultId) -> R) {
...
-    test_with(Token(DOT), Token(KBTC), None);
-    test_with(Token(DOT), Token(IBTC), None);
-    test_with(Token(DOT), Token(IBTC), Some(Token(KSM)));
-    test_with(Token(KSM), Token(IBTC), None);
-    test_with(ForeignAsset(1), Token(IBTC), None);
+    // test_with(Token(DOT), Token(KBTC), None);
+    // test_with(Token(DOT), Token(IBTC), None);
+    // test_with(Token(DOT), Token(IBTC), Some(Token(KSM)));
+    // test_with(Token(KSM), Token(IBTC), None);
+    // test_with(ForeignAsset(1), Token(IBTC), None);
```
Comment (gregdhill) [inline standalone/runtime/tests/test_redeem.rs:42]: Uncomment

File: crates/vault-registry/src/lib.rs [hunk 2]
```diff
@@ -872,6 +872,9 @@ impl<T: Config> Pallet<T> {
         // will fail if free_balance is insufficient
         amount.lock_on(&vault_id.account_id)?;
 
+        // withdraw first such that past rewards don't get changed by this deposit
+        ext::fee::withdraw_all_vault_rewards::<T>(vault_id)?;
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:876]: Why is this necessary? Nomination does the same thing to prevent new nominators earning past rewards but I'm not sure why this matters here

**Review by gregdhill — CHANGES_REQUESTED** [review]

File: crates/vault-registry/src/lib.rs (hunk 2 above)
Comment (sander2) [reply crates/vault-registry/src/lib.rs:876]: It's a change brought to light by my randomized tests, so I was quite happy about that :)

To see why this is necessary, consider this scenario:
- vault deposits 1 dot collateral
- nominator deposits 1 dot collateral
- 1 INTR rewards are distributed. Both vault and nominator should get 50% (ignoring commission). But we don't withdraw yet.
- vault deposits 1 additional dot
- Now we withdraw from the reward pool to the staking pool. With the current stakes, vault gets 2/3rd, nominator gets 1/3rd --> wrong!

Comment (gregdhill) [reply crates/vault-registry/src/lib.rs:876]: Good catch! Makes sense to me, thanks for the example
